walkthroughs/week-2-server/examples/server-stats/src/main/java/com/google/sps/servlets/ServerStatsServlet.java

```diff
@@ -48,20 +48,9 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
    * Converts a ServerStats instance into a JSON string using manual String concatentation.
    */
   private String convertToJson(ServerStats serverStats) {
-    String json = "{";
-    json += "\"startTime\": ";
-    json += "\"" + serverStats.getStartTime() + "\"";
-    json += ", ";
-    json += "\"currentTime\": ";
-    json += "\"" + serverStats.getCurrentTime() + "\"";
-    json += ", ";
-    json += "\"maxMemory\": ";
-    json += serverStats.getMaxMemory();
-    json += ", ";
-    json += "\"usedMemory\": ";
-    json += serverStats.getUsedMemory();
-    json += "}";
-    return json;
+    return String.format("{\"startTime\": \"%s\", \"currentTime\": \"%s\", \"maxMemory\": %d, \"usedMemory\": %d}",
```

Comment:
I wanted to break it the string down into very obvious steps, to make it more obvious what the JSON format was, since readers will still be brand new to it. But I agree that `String.format()` is a little easier on the eyes.

What do you think about making this a comment? Something like:

```
// We could also use String.format() here:
// return String.format("{\"startTime\": \"%s\", \"currentTime\": \"%s\", \"maxMemory\": %d, \"usedMemory\": %d}",
// serverStats.getStartTime(), serverStats.getCurrentTime(),
// serverStats.getMaxMemory(), serverStats.getUsedMemory());
```

Reply:
A comment just makes the code more cluttered/distracting. Keep one or the other.
I like String.format(..) since it's what I'd expect to see in "real" code, and is a common pattern across virtually any programming language.

Your call, though. The step-by-step construction works just fine

Reply:
I appreciate your input, and I think I'm going to stick with the step-by-step approach. Part of my goal was to show that this isn't a great way to do it, to encourage readers to use Gson instead.